### PR TITLE
Adding MPLS changes and VLAN mask

### PR DESCRIPTION
**Pull request by karamsivia**
Adding MPLS changes and VLAN mask

### CODE REVIEW

File: openflow13/action.go [hunk 1]
```diff
@@ -311,6 +311,36 @@ func (a *ActionPush) UnmarshalBinary(data []byte) error {
     return nil
 }
 
+func NewActionPopMpls(etherType uint16) *ActionPopMpls {
```
Comment (shaleman) [inline openflow13/action.go]: minor: Can we move these functions below ActionPopMpls structure definitions?

File: openflow13/match.go [hunk 2]
```diff
@@ -473,7 +475,7 @@ func (m *VlanIdField) UnmarshalBinary(data []byte) error {
 }
 
 // Return a MatchField for vlan id matching
-func NewVlanIdField(vlanId uint16) *MatchField {
+func NewVlanIdField(vlanId uint16, mask bool) *MatchField {
```
Comment (shaleman) [inline openflow13/match.go]: mask needs to be *uint16 type. see NewEthSrcField() for example of how to use masks

File: openflow13/match.go [hunk 3]
```diff
@@ -483,7 +485,84 @@ func NewVlanIdField(vlanId uint16) *MatchField {
 	vlanIdField.VlanId = vlanId | OFPVID_PRESENT
 	f.Value = vlanIdField
 	f.Length = uint8(vlanIdField.Len())
+	
+	if mask == true { 
+		mask := new(VlanIdField)
+		mask.VlanId = OFPVID_PRESENT
+		f.Mask = mask
```
Comment (shaleman) [inline openflow13/match.go:492]: This is not correct. Please see NewEthSrcField() for an example of how to use points to pass the mask

**Comment by shaleman** [comment]
@karamsivia 
Changes look fine other than the clan id mask handling

**Comment by karamsivia** [comment]
@shaleman 
Based on the feedback, changed Vlan mask handling. Also moved popmpls related functions below ActionPopMpls structure definitions

**Comment by shaleman** [comment]
LGTM
Thanks for making the change

**Pull request merged**